### PR TITLE
Add handle flush for lite API and update cmake build

### PR DESCRIPTION
**Pull request by ZongwuYang**
Signed-off-by: Zongwu Yang <zongwu.yang@verisilicon.com>

* Add handle flush for lite API
* Add cmake build for lite API
* Update tim-vx link type as PUBLIC to fix app dependence to driver libs when building app

### CODE REVIEW

File: include/tim/lite/handle.h [hunk 1]
```diff
@@ -32,6 +32,11 @@ namespace lite {
...
 class HandleImpl;
 
+enum class HandleFlushType {
+    HandleFlush = 0,
+    HandleInvalidate = 1
+};
+
 class Handle {
     public:
         std::unique_ptr<HandleImpl>& impl() { return impl_; }
```
Comment (sunshinemyson) [inline include/tim/lite/handle.h:42]: Shall we add Flush/Invalidate API for handle? It's easy to understand.

File: include/tim/lite/execution.h [hunk 2]
```diff
@@ -40,6 +40,7 @@ class Execution {
         virtual Execution& BindInputs(const std::vector<std::shared_ptr<Handle>>& handles) = 0;
         virtual Execution& BindOutputs(const std::vector<std::shared_ptr<Handle>>& handles) = 0;
         virtual bool Trigger() = 0;
+        virtual bool HandleFlush(const std::shared_ptr<Handle>& handle, HandleFlushType type) = 0;
```
Comment (sunshinemyson) [inline include/tim/lite/execution.h:43]: I prefer add separate API in handle interface.

**Review by sunshinemyson — CHANGES_REQUESTED** [review]

File: include/tim/lite/handle.h [hunk 3]
```diff
@@ -32,9 +32,16 @@ namespace lite {
 
 class HandleImpl;
 
+enum class HandleFlushType {
+    HandleFlush = 0,
+    HandleInvalidate = 1
+};
+
```
Comment (sunshinemyson) [inline include/tim/lite/handle.h:39]: If you want to keep this difinition, then move them to src not in the public api.

Comment (ZongwuYang) [reply include/tim/lite/handle.h]: Have updated.

**Review by sunshinemyson — APPROVED** [review]

**Pull request merged**